### DISCUSSION TITLE
What difference between kyuubi.session.idle.timeout and kyuubi.session.engine.idle.timeout ?

### DISCUSSION THREAD

**Discussion by ethanzhou8109** [apache/kyuubi#3086] [Q&A] [answered]
Is session.idle.timeout better to be longer than engine.idle.timeout ?

**Comment by wForget** [comment, marked as answer]
The session will be closed when not accessed for more than the `kyuubi.session.idle.timeout` and the engine will be closed when not accessed for more than the `kyuubi.engine.idle.timeout`.
So we only need to understand the relationship between session and engine, they are not one-to-one correspondence. For the session, we usually use it to close it after the use is complete, and the `kyuubi.session.idle.timeout` only handles some abnormal sessions. The CONNECTION level engine stops when the session is closed, and the shared engine will judge whether it needs to exit through `kyuubi.engine.idle.timeout`.
